client/http/http.go

```diff
@@ -7,31 +7,64 @@ import (
 	"context"
 	"io"
 	"net/http"
+	"strconv"
 	"time"
 
+	"github.com/opentracing-contrib/go-stdlib/nethttp"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	"github.com/prometheus/client_golang/prometheus"
+
 	"github.com/beatlabs/patron/correlation"
 	"github.com/beatlabs/patron/encoding"
 	"github.com/beatlabs/patron/log"
 	"github.com/beatlabs/patron/reliability/circuitbreaker"
 	"github.com/beatlabs/patron/trace"
-	"github.com/opentracing-contrib/go-stdlib/nethttp"
-	opentracing "github.com/opentracing/opentracing-go"
-	"github.com/opentracing/opentracing-go/ext"
 )
 
 const (
 	clientComponent = "http-client"
 )
 
+var (
+	reqTotalMetric   *prometheus.CounterVec
+	reqLatencyMetric *prometheus.HistogramVec
+)
+
+func init() {
+	reqTotalMetric = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "client",
+			Subsystem: "http",
+			Name:      "requests_total",
+			Help:      "Total number of HTTP requests sent by the client.",
+		},
+		[]string{"method", "url", "status_code"},
+	)
+	prometheus.MustRegister(reqTotalMetric)
+
+	reqLatencyMetric = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "client",
```

Comment:
I would suggest to have the following (as discussed):
Namespace: name of the service
Subsystem: http_client

Reply:
As discussed, having the service name prefix would be ideal, just not entirely sure how to accomplish here.
I'll look into it.

Reply:
As discussed offline with @mantzas:
There are other Prometheus labels one could use to narrow down the metrics for a specific service. 
There could be cases when one needs to aggregate metrics for many services, e.g. per namespace.
Most of the Patron server-side components don't use the service name prefix.

Adding the service name prefix here introduces quite a bit of noise in code and doesn't really offer much.

Let's keep as is.